src/init.cpp

```diff
@@ -1678,13 +1678,24 @@ bool AppInitMain(InitInterfaces& interfaces)
 
     // ********************************************************* Step 11: import blocks
 
-    if (!CheckDiskSpace(GetDataDir())) {
-        InitError(strprintf(_("Error: Disk space is low for %s"), GetDataDir()));
-        return false;
-    }
-    if (!CheckDiskSpace(GetBlocksDir())) {
-        InitError(strprintf(_("Error: Disk space is low for %s"), GetBlocksDir()));
-        return false;
+    {
+        LOCK(cs_main);
+        uint64_t additional_bytes_needed = 0;
+        if (!fReindex && chainActive.Height() <= 1) {
+            // If first startup
+            if (fPruneMode)
+                additional_bytes_needed = nPruneTarget;
+            else
+                additional_bytes_needed = chainparams.AssumedBlockchainSize() * 1024 * 1024 * 1024;
+        }
+        if (!CheckDiskSpace(GetDataDir())) {
+            InitError(strprintf(_("Error: Disk space is low for %s"), GetDataDir()));
+            return false;
+        }
```

Comment:
May as well leave this check out of the locked section, as it has no need for the related info.

Reply:
Thank you for the review. I tested outside of the lock and it makes bitcoind hanging forever on error.